vmm/src/vm.rs

```diff
@@ -590,6 +596,22 @@ impl DeviceManager {
             )?;
         }
 
+        // Add virtio-console if required
+        if vm_cfg.console.enabled {
+            let virtio_console_device = vm_virtio::Console::new(Some(Box::new(stdout())))
```

Comment:
Here the creation of the console should return a `virtio_console` and a `console_input`. The `virtio_console` can be given to the VirtioPciDevice, while the `console_input` should be given to the `DeviceManager` to let it call into `queue_input_bytes()`.

Reply:
Done by @sboeuf